app/models/concerns/orchestration/dns.rb

```diff
@@ -13,163 +15,199 @@ def dns?
     (host.nil? || host.managed?) && managed? && hostname.present? && ip_available? && !domain.nil? && !domain.proxy.nil? && SETTINGS[:unattended]
   end
 
+  def dns6?
+    # host.managed? and managed? should always come first so that orchestration doesn't
+    # even get tested for such objects
+    (host.nil? || host.managed?) && managed? && hostname.present? && ip6_available? && !domain.nil? && !domain.proxy.nil? && SETTINGS[:unattended]
```

Comment:
Looks like
`SETTINGS[:unattended] && (host.nil? || host.managed?) && managed? && hostname.present?` is repeated quite a lot, how about extracting that to a separate method? (and checking the settings before all the other checks)